src/Microsoft.Framework.Runtime.Roslyn/RoslynProjectReference.cs

```diff
@@ -197,14 +197,22 @@ public IDiagnosticResult EmitAssembly(string outputPath)
 
                 Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
 
-                var afterCompileContext = new AfterCompileContext(CompilationContext, emitResult.Diagnostics)
+                var afterCompileContext = new AfterCompileContext()
                 {
+                    ProjectContext = BeforeCompileContext.ProjectContext,
+                    Compilation = BeforeCompileContext.Compilation,
                     AssemblyStream = assemblyStream,
                     SymbolStream = pdbStream,
-                    XmlDocStream = xmlDocStream
+                    XmlDocStream = xmlDocStream,
+                    Diagnostics = new List<Diagnostic>()
```

Comment:
I think there is overload for this constructor that takes an `IEnumerable`. Then, you don't need to `foreach` below